### DISCUSSION TITLE
Support private gitlab repo as git submodule

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#32703] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

I'm sure I mentioned it in the past in one of the discussions.
According to my experience, this use case is not supported. I'm unable to generate a reproduction repo as the the referenced gitlab repos are private.
I have reason to believe that gitlab supports the `insteadOf` feature.  Could you please advice how to contribute a fix?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
It's been a long time since I looked into this code.

Are you using `cloneSubmodules`, or just wanting Renovate to be able to update the submodule digest using the git-submodules manager?

**Reply by lmilbaum** [reply]
I am using `cloneSubmudles`. The clone step is failing due to not having permissions.

**Comment by rarkins** [comment]
While supporting another `cloneSubmodules` support request I came across our docs which say:

> Important: private submodules aren't supported by Renovate, unless the underlying ssh layer already has the correct permissions.

This implies that it _can_ work with SSH auth.

Any code relating to git.insteadOf would need to go in https://github.com/renovatebot/renovate/blob/10f1f46d382e83dbc92140186b2e4fbf972ea645/lib/util/git/index.ts#L348-L378

**Reply by lmilbaum** [reply]
Thanks @rarkins. I'll look into it.
